index.bs

```diff
@@ -3209,6 +3210,81 @@ opened</dfn> steps given |window|, |type| and |message|.
 
 </div>
 
+## Emulation ## {#module-emulation}
+
+The <dfn export for=modules>emulation</dfn> module contains functionality for
+emulating device-dependent and system-dependent parameters.
+
+### Definition ### {#module-emulation-definition}
+
+[=remote end definition=]
+
+<pre class="cddl remote-cddl">
+
+EmulationCommand = (
+  emulation.SetViewport
+)
+</pre>
+
+### Types ### {#module-emulation-types}
+
+### Commands ### {#module-emulation-commands}
+
+#### The emulation.setViewport Command ####  {#command-emulation-setViewport}
+
+The <dfn export for=commands>emulation.setViewport</dfn> command emulates specific viewport characteristics (e.g. viewport width and viewport height) on the given top-level browsing context.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+    <pre class="cddl remote-cddl">
+      emulation.SetViewport = {
+        method: "emulation.setViewport",
```

Comment:
Are we also going to end up adding DPI settings to the same command? It seems like from a protocol efficiency point of view, having a single command for all the things that people might want to set at the same time might be an improvement.

Reply:
I would add device pixel ratio as a separate command (if we ever define it) as it can be used to test things independently from the viewport. E.g., one could test the media query for dpr without changing the viewport. Would the ability to send commands in batch help here?

Reply:
I have updated it to reset the viewport with a null value. So it looks like even if we end up needing the DPI settings in this command we would be able to extend?